source/async-return-type.d.ts

```diff
@@ -0,0 +1,19 @@
+import {PromiseValue} from './promise-value';
+
+type AsyncFunction = (...args: unknown[]) => Promise<unknown>;
+
+/**
+Unpack the return type of a function that returns a `Promise`.
+
```

Comment:
> Write about some real-world use-cases where it can be useful. (It can be hard sometimes for users to see where they would use something) - https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md

and

> If there has been any discussion somewhere about this type, include a link to it. For example, a discussion on the TypeScript issue tracker.